src/components/Layout/index.tsx

```diff
@@ -39,22 +40,55 @@ const HEADER_LINKS = [
   }
 ];
 
+const HUB_LINK = {
+  href: '/hub',
+  text: 'Member hub'
+};
+
+type MembershipStatus = 'pending' | 'active' | null;
+
 const Layout: React.FC<Props> = ({
   children,
   title,
   description,
   hideNewsletter
 }) => {
+  const [links, setLinks] = useState(JSON.stringify(HEADER_LINKS));
+
+  const onMembershipFetched = (response: { status: MembershipStatus }) => {
```

Comment:
It is still not clear for me which is the contract to consume. Here we should add the expected response from the server and consume it properly @orlando